packages/component/src/provider.ts

```diff
@@ -250,9 +252,44 @@ export class Provider<T extends Name> {
         instanceDeferred.resolve(instance);
       }
     }
+
+    this.invokeOnInitCallbacks(instance, normalizedIdentifier);
+
     return instance;
   }
 
+  /**
+   *
+   * @param callback - a function that will be invoked  after the provider has been initialized by calling provider.initialize().
+   * The function is invoked SYNCHRONOUSLY, so it should not execute any longrunning tasks in order to not block the program.
+   *
+   * @returns a function to unregister the callback
+   */
+  onInit(callback: (instance: NameServiceMapping[T]) => void): () => void {
```

Comment:
`callback` is of type `OnInitCallback<T>` right? Can that just be put here instead of the function signature? I notice `OnInitCallback` has an `identifier` param this one doesn't have, is that intentional?

Reply:
Good catch. Updated to use the type alias